src/Avalonia.Controls/Control.cs

```diff
@@ -403,7 +403,7 @@ protected override void OnGotFocus(GotFocusEventArgs e)
                 {
                     if (_focusAdorner == null)
                     {
-                        var template = GetValue(FocusAdornerProperty);
+                        var template = GetValue(FocusAdornerProperty) ?? adornerLayer.DefaultFocusAdorner;
```

Comment:
Does `IsSet()` have any perf advantages here instead of the property read? I've seen it used recently in other stuff.

Reply:
I wouldn't say so https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/PropertyStore/ValueStore.cs#L252

Reply:
Only difference - it doesn't check for inherited properties (not enabled here) and doesn't create default value (not set here).